frontend/model/contracts/chatroom.js

```diff
@@ -191,16 +186,14 @@ sbp('chelonia/defineContract', {
       process ({ data, meta, hash, id }, { state }) {
         const { username } = data
         if (!state.onlyRenderMessage && state.users[username]) {
-          // this can happen when we're logging in on another machine, and also in other circumstances
-          console.warn('Can not join the chatroom which you are already part of')
-          return
+          throw new Error(`Can not join the chatroom which ${username} is already part of`)
         }
 
         Vue.set(state.users, username, { joinedDate: meta.createdDate })
 
         const { type, privacyLevel } = state.attributes
-        const isPrivateDM = type === CHATROOM_TYPES.INDIVIDUAL && privacyLevel === CHATROOM_PRIVACY_LEVEL.PRIVATE
-        if (!state.onlyRenderMessage || isPrivateDM) {
+        const isDirectMessage = type === CHATROOM_TYPES.INDIVIDUAL && privacyLevel === CHATROOM_PRIVACY_LEVEL.PRIVATE
+        if (!state.onlyRenderMessage || isDirectMessage) {
```

Comment:
Something doesn't make sense conceptually here.

1. It is not possible for DMs to have `CHATROOM_TYPES.GROUP`.
2. `CHATROOM_TYPES.INDIVIDUAL` is unclear, it should be renamed to `CHATROOM_TYPES.DIRECT_MESSAGE`
3. All `CHATROOM_TYPES.DIRECT_MESSAGE` have `CHATROOM_PRIVACY_LEVEL.PRIVATE`. It doesn't need to even be checked, because there is no such thing as a "group" DM or a "public" DM.
4. `CHATROOM_PRIVACY_LEVEL` is only relevant for `CHATROOM_TYPES.GROUP`

Can you update the code to reflect this?